src/test/ui/issues/issue-37515.stderr

```diff
@@ -2,8 +2,10 @@ warning: type alias is never used: `Z`
   --> $DIR/issue-37515.rs:5:1
    |
 LL | type Z = dyn for<'x> Send;
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_Z`
```

Comment:
The structured suggestions are slightly incorrect. We want to point only at `Z`, not the whole item. I believe this is a problem for enum variants with data and consts as well.

Reply:
I was able to get it to look like this, is this sufficient? I'm passing in the right span now but it's still printing the `^` underline.

```
warning: type alias is never used: `Z`
  --> $DIR/issue-37515.rs:5:1
   |
LL | type Z = dyn for<'x> Send;
   | ^^^^^-^^^^^^^^^^^^^^^^^^^^
   |      |
   |      help: if this is intentional, prefix it with an underscore: `_Z`
   |
   = note: the leading underscore signals that this type alias serves some other purpose even if it isn't used in a way that we can detect.
```